From 1d78555bfc753039546ce5ca655dece67732e7bd Mon Sep 17 00:00:00 2001
From: Mark Corbin <mark.corbin@embecosm.com>
Date: Thu, 22 Nov 2018 12:19:11 +0000
Subject: [PATCH] gcc: define _REENTRANT for RISC-V when -pthread is passed

The detection of pthread support fails on RISC-V unless _REENTRANT
is defined. Added the CPP_SPEC definition from gcc 8.1.0 to correct
this.

Signed-off-by: Mark Corbin <mark.corbin@embecosm.com>
Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 gcc/config/riscv/linux.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/config/riscv/linux.h b/gcc/config/riscv/linux.h
index 4b2f7b6e1fd..b00d23ddfa0 100644
--- a/gcc/config/riscv/linux.h
+++ b/gcc/config/riscv/linux.h
@@ -47,6 +47,8 @@ along with GCC; see the file COPYING3.  If not see
 
 #define ICACHE_FLUSH_FUNC "__riscv_flush_icache"
 
+#define CPP_SPEC "%{pthread:-D_REENTRANT}"
+
 #define LINK_SPEC "\
 -melf" XLEN_SPEC "lriscv \
 %{shared} \
-- 
2.21.0

